runtime/lua/vim/secure.lua

```diff
@@ -0,0 +1,102 @@
+local M = {}
+
+--- Attempt to read the file at {path} prompting the user if the file should be
+--- trusted. The user's choice is persisted in a trust database at
+--- $XDG_STATE_HOME/nvim/trust.
+---
+---@param path (string) Path to a file to read.
+---
+---@return (string|nil) The contents of the given file if it is trusted, or nil
+---                     otherwise.
+function M.read(path)
+  vim.validate({ path = { path, 's' } })
+  local fullpath = vim.fn.fnamemodify(vim.fs.normalize(path), ':p')
+  local trust = {}
+  do
+    local f = io.open(vim.fn.stdpath('state') .. '/trust', 'r')
+    if f then
+      local contents = f:read('*a')
+      if contents then
+        for line in vim.gsplit(contents, '\n') do
+          local hash, file = string.match(line, '^(%S+) (.+)$')
```

Comment:
Is this actually necessary? The input path argument is already expanded to a full (absolute) path, so if a relative path exists here it would never compare equal against the absolute input path.

Reply:
Yes, you're right, it's not necessary. What worries me however is that `:p`'s behavior in `fnamemodify` is "unpredictable" if file doesn't exist, and that could potentially lead to something unexpected. Can you replace that with `fs_realpath`? `fs_realpath` fails when file doesn't exist

Reply:
Fixed in 7578fc69f2f9940b36b019a7d3fe0729302890e6. I think it might make sense to also use `fs_realpath` in `vim.fs.normalize()`. If we make that change then we can remove it here.